src/views/Header.js

```diff
@@ -80,6 +81,17 @@ class Header extends React.PureComponent<void, HeaderProps, HeaderState> {
     ).options.headerTruncatedBackTitle;
   }
 
+  _getBackButtonTitleStyle(scene: NavigationScene): ?Style {
+    const lastScene = this._getLastScene(scene);
+    const { headerBackTitleStyle } = this.props.getScreenDetails(scene).options;
+    if (!lastScene) {
+      return headerBackTitleStyle;
+    }
+    const { headerTitleStyle } = this.props.getScreenDetails(lastScene).options;
+
+    return [headerTitleStyle, headerBackTitleStyle];
```

Comment:
>  I want to see an app with different styles (e.g. different font family) between two screens

I actually don't have any example of such an iOS app.

> also the back button doesn't exactly transition from the title, there seems to be a disconnect between them.

Not exactly, but it fades over, making it look like seamless. But making the transition/animation like the native one is a different issue I guess.